### ISSUE TITLE
azurerm_app_service incorrect documentation regarding sas_url

### ISSUE THREAD

**Issue by mleziva** [terraform-providers/terraform-provider-azurerm#8872]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Overview
The documentation for azurerm_app_service states that there is no way to get a storage account sas token using terraform, but there is a way using azurerm_storage_account_sas available here: https://www.terraform.io/docs/providers/azurerm/d/storage_account_sas.html

### Terraform (and AzureRM Provider) Version

AzureRM 2.27.0
Terraform 0.12.20

### Affected Resource(s)

* `azurerm_app_service`

### Terraform Configuration Files

```hcl
data "azurerm_storage_account_sas" "diagnostics_storage_account_sas" {
  connection_string = data.azurerm_storage_account.diagnostics_storage_account.primary_connection_string
  https_only        = true

  resource_types {
    service   = false
    container = true
    object    = true
  }

  services {
    blob  = true
    queue = false
    table = true
    file  = false
  }

  start  = timestamp()
  expiry = timeadd(timestamp(), "8760h")

  permissions {
    read    = true
    write   = true
    delete  = false
    list    = true
    add     = true
    create  = true
    update  = true
    process = true
  }
}
```

### Important Factoids

Incorrect page: https://www.terraform.io/docs/providers/azurerm/r/app_service.html

Variable description:
sas_url - (Required) The URL to the storage container, with a Service SAS token appended. NOTE: there is currently no means of generating Service SAS tokens with the azurerm provider.

Requested fix: Replace the NOTE text with a link to https://www.terraform.io/docs/providers/azurerm/d/storage_account_sas.html

### References

https://www.terraform.io/docs/providers/azurerm/r/app_service.html
* #0000

**Comment by HorizonNet** [comment]
I think that the description is still accurate. It mentions that there's no way of generating the SAS token. The link for the fix is pointing to a data source and not to a resource. It would take an existing SAS, but still the provider is not able to generate it. The description also talks about the Service SAS. The data source is an Account SAS.

**Comment by mleziva** [comment]
While you are technically correct, the issue is that the documentation is misleading. If someone who is not familiar with AzureRM simply reads that description they may conclude that there is no means of obtaining a SAS token using Terraform. The `sas_url `can be any SAS token regardless of how it is generated which includes service or account generated SAS tokens.

**Comment by tombuildsstuff** [comment]
Fixed via #9438
